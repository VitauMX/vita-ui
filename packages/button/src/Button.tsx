import * as React from 'react';
import styled from 'styled-components';
import { variant } from 'styled-system';
import buttonBaseStyles from './ButtonStyles';

const variants = variant({
  variants: {
    primary: {
      color: '#FFFFFF',
      borderWidth: '0px',
      padding: '0px 32px',
      fontFamily: 'Open Sans',
      backgroundColor: '#0099E6',
      '&:hover:enabled': {
        backgroundColor: '#0088CC',
      },
      '&:active:enabled': {
        backgroundColor: '#0077B3',
      },
      '&:focus:enabled': {
        boxShadow: '0px 0px 0px 2px #80D4FF',
      },
      '&:disabled': {
        cursor: 'not-allowed',
        backgroundColor: '#80D4FF',
      },
    },
  },
});

const destructive = variant({
  prop: 'destructive',
  variants: {
    true: {
      backgroundColor: '#FF2D1A',
      '&:hover:enabled': {
        backgroundColor: '#CC1100',
      },
      '&:active:enabled': {
        backgroundColor: '#B30F00',
      },
      '&:focus:enabled': {
        boxShadow: '0px 0px 0px 2px #FF8A7F',
      },
      '&:disabled': {
        cursor: 'not-allowed',
        backgroundColor: '#FF5B4C',
      },
    },
  },
});

const size = variant({
  prop: 'size',
  variants: {
    large: {
      height: '56px',
    },
    small: {
      padding: '0px 16px',
      height: '32px',
    },
  },
});

export interface IButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  variant: 'primary' | 'secondary';
  destructive?: boolean;
  size?: 'large' | 'small';
  disabled?: boolean;
}

const StyledButton = styled.button`
  ${buttonBaseStyles}
  ${variants}
  ${destructive}
  ${size}
`;

export const Button = React.forwardRef<HTMLButtonElement, IButtonProps>(
  (props, ref) => {
    return <StyledButton ref={ref} disabled={props.disabled} {...props} />;
  }
);

Button.displayName = 'Button';
