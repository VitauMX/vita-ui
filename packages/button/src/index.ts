export * from './ButtonPrimary';
export * from './ButtonTertiary';
export * from './ButtonSecondary';
export * from './ButtonText';
