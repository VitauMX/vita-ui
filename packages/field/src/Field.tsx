import * as React from 'react';
import styled from 'styled-components';
import { useId } from '@reach/auto-id';

const StyledFieldContainer = styled.div`
  position: relative;
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  color: #4f6672;
`;

const StyledFieldLabel = styled.label`
  position: absolute;
  padding: 16px 20px;
  height: 100%;
  left: 0;
  top: 0;
  transform-origin: 0 0;
  cursor: text;
  pointer-events: none;
  font-weight: 600;
  transition: 100ms all ease-in;
  border: 1px solid transparent;
  max-width: 66.66%;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

const StyledInput = styled.input`
  height: 54px;
  padding: 0px 20px;
  min-width: 220px;
  width: 100%;
  line-height: 22px;
  letter-spacing: 0.01em;
  background-color: #fff;
  border: 1px solid #bac4ca;
  border-radius: 4px;
  appearance: none;
  box-sizing: border-box;
  transition: 100ms all ease-in;

  &:read-only,
  &:disabled {
    cursor: not-allowed;
    opacity: 1;
  }

  &:disabled {
    background-color: #f1f3f4;
  }

  &::placeholder {
    opacity: 0;
    transition: inherit;
  }

  &:hover {
    border-color: #8698a2;
  }

  &:focus {
    outline: 0;
    border-color: #1ab3ff;
    box-shadow: 0 0 0 2px #b3e5ff;

    &::placeholder {
      opacity: 1;
    }
  }

  &:hover,
  &:focus {
    border-width: 2px;
    & ~ ${StyledFieldLabel} {
      border-width: 2px;
    }
  }

  &:focus,
  &:not(:placeholder-shown) {
    padding-top: 24px;
    padding-bottom: 8px;

    & ~ ${StyledFieldLabel} {
      font-size: 12px;
      padding-top: 8px;
      padding-bottom: 30px;
    }
  }

  /* Fix for autofill, see https://stackoverflow.com/questions/55427966/inputnotplaceholder-shown-label-selector-does-not-work-with-autofill */
  &:-webkit-autofill {
    padding-top: 24px;
    padding-bottom: 8px;

    & ~ ${StyledFieldLabel} {
      font-size: 12px;
      padding-top: 8px;
      padding-bottom: 30px;
    }
  }
`;

export interface FieldProps
  extends React.InputHTMLAttributes<HTMLInputElement> {
  label: string;
}

export const Field = React.forwardRef<HTMLInputElement, FieldProps>(
  (props, ref) => {
    const {
      id,
      label,
      /*
        Must always have a placeholder, see
        https://www.samanthaming.com/tidbits/88-css-placeholder-shown/#placeholder-shown-must-have-placeholder
      */
      placeholder = ' ',
      ...inputProps
    } = props;

    const innerId = useId(id);

    return (
      <StyledFieldContainer>
        <StyledInput
          ref={ref}
          {...inputProps}
          placeholder={placeholder}
          id={innerId}
        />
        <StyledFieldLabel htmlFor={innerId}>{label}</StyledFieldLabel>
      </StyledFieldContainer>
    );
  }
);

Field.displayName = 'Field';
