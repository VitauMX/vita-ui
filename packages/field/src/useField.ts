import * as React from 'react';
import { useId } from '@reach/auto-id';

export interface FieldProps
  extends React.InputHTMLAttributes<HTMLInputElement> {
  label: string;
}

interface UseFieldPropGetters {
  inputProps: React.InputHTMLAttributes<HTMLInputElement>;
  labelProps: React.LabelHTMLAttributes<HTMLLabelElement>;
}

export function useField(props: FieldProps): UseFieldPropGetters {
  const { id, label: _, ...restInputProps } = props;

  const innerId = useId(id);

  return {
    inputProps: {
      ...restInputProps,
      id: innerId,
      /*
        Must always have a placeholder, see
        https://www.samanthaming.com/tidbits/88-css-placeholder-shown/#placeholder-shown-must-have-placeholder
      */
      placeholder: restInputProps.placeholder || ' ',
    },
    labelProps: {
      htmlFor: innerId,
    },
  };
}
