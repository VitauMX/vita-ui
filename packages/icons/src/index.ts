export * from './Icon';
export * from './createIcon';
export * from './icons/';
