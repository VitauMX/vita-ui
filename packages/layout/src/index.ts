export * from './Grid';
export * from './Box';
export * from './Flex';
