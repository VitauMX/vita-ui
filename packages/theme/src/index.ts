export * from './ThemeProvider';
export * from './theme';
export * from './utils';
