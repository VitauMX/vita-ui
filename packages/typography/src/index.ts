export * from './Body';
export * from './Caption';
export * from './Label';
export * from './Message';
export * from './Title';
export * from './Overline';
