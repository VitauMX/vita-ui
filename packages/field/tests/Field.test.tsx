import React from 'react';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';

import { Field } from '../src';

describe('Field', () => {
  test('renders correctly', () => {
    const { container } = render(<Field label="Basic field" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have no axe violations', async () => {
    const { container } = render(<Field label="Basic field" />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('passes ref to underlying DOM element', () => {
    const ref = React.createRef<HTMLInputElement>();
    const label = 'Basic field';
    const { getByLabelText } = render(<Field ref={ref} label={label} />);

    expect(getByLabelText(label)).toBe(ref.current);
  });

  test('should render with disabled', () => {
    const { container } = render(
      <Field label="Basic field" defaultValue="Value" disabled />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render with success', () => {
    const { container } = render(
      <Field label="Basic field" defaultValue="Value" validation="success" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render with error', () => {
    const { container } = render(
      <Field label="Basic field" defaultValue="Value" validation="error" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
