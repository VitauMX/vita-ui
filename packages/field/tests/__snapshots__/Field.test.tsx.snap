// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field renders correctly 1`] = `
.c2 {
  position: absolute;
  padding: 16px 20px;
  height: 100%;
  left: 0;
  top: 0;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
  cursor: text;
  pointer-events: none;
  font-weight: 600;
  -webkit-transition: 100ms all ease-in;
  transition: 100ms all ease-in;
  border: 1px solid transparent;
  max-width: 66.66%;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: var(--field-label-color);
}

.c1 {
  height: 54px;
  padding: 0px 20px;
  min-width: 220px;
  width: 100%;
  line-height: 22px;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  background-color: #fff;
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  box-sizing: border-box;
  -webkit-transition: 100ms all ease-in;
  transition: 100ms all ease-in;
  color: #4f6672;
  border-color: var(--field-input-border-color);
}

.c1:-moz-read-only,
.c1:disabled {
  cursor: not-allowed;
  opacity: 1;
}

.c1:read-only,
.c1:disabled {
  cursor: not-allowed;
  opacity: 1;
}

.c1:disabled {
  background-color: #f1f3f4;
}

.c1::-webkit-input-placeholder {
  opacity: 0;
  -webkit-transition: inherit;
  transition: inherit;
}

.c1::-moz-placeholder {
  opacity: 0;
  -webkit-transition: inherit;
  transition: inherit;
}

.c1:-ms-input-placeholder {
  opacity: 0;
  -webkit-transition: inherit;
  transition: inherit;
}

.c1::placeholder {
  opacity: 0;
  -webkit-transition: inherit;
  transition: inherit;
}

.c1:hover {
  border-color: var(--field-input-hover-border-color);
}

.c1:focus {
  outline: 0;
  border-color: var(--field-input-focus-border-color);
  box-shadow: var(--field-input-box-shadow);
}

.c1:focus::-webkit-input-placeholder {
  opacity: 1;
}

.c1:focus::-moz-placeholder {
  opacity: 1;
}

.c1:focus:-ms-input-placeholder {
  opacity: 1;
}

.c1:focus::placeholder {
  opacity: 1;
}

.c1:hover,
.c1:focus {
  border-width: 2px;
}

.c1:hover ~ label,
.c1:focus ~ label {
  border-width: 2px;
}

.c1:focus,
.c1:not(:placeholder-shown) {
  padding-top: 24px;
  padding-bottom: 8px;
}

.c1:focus ~ label,
.c1:not(:placeholder-shown) ~ label {
  color: var(--field-label-focus-color);
  font-size: 12px;
  padding-top: 8px;
  padding-bottom: 30px;
}

.c1:-webkit-autofill {
  padding-top: 24px;
  padding-bottom: 8px;
}

.c1:-webkit-autofill ~ label {
  color: var(--field-label-focus-color);
  font-size: 12px;
  padding-top: 8px;
  padding-bottom: 30px;
}

.c0 {
  position: relative;
  font-family: 'Open Sans',sans-serif;
  font-size: 16px;
  --field-input-border-color: #bac4ca;
  --field-input-hover-border-color: #8698a2;
  --field-input-focus-border-color: #1ab3ff;
  --field-input-box-shadow: 0 0 0 2px #b3e5ff;
  --field-label-color: #4f6672;
  --field-label-focus-color: #4f6672;
}

<div
  class="c0"
>
  <input
    aria-labelledby="2"
    class="c1"
    id="1"
    placeholder=" "
  />
  <label
    class="c2"
    for="1"
    id="2"
  >
    Basic field
  </label>
</div>
`;

exports[`Field should render with disabled 1`] = `
.c2 {
  position: absolute;
  padding: 16px 20px;
  height: 100%;
  left: 0;
  top: 0;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
  cursor: text;
  pointer-events: none;
  font-weight: 600;
  -webkit-transition: 100ms all ease-in;
  transition: 100ms all ease-in;
  border: 1px solid transparent;
  max-width: 66.66%;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: var(--field-label-color);
}

.c1 {
  height: 54px;
  padding: 0px 20px;
  min-width: 220px;
  width: 100%;
  line-height: 22px;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  background-color: #fff;
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  box-sizing: border-box;
  -webkit-transition: 100ms all ease-in;
  transition: 100ms all ease-in;
  color: #4f6672;
  border-color: var(--field-input-border-color);
}

.c1:-moz-read-only,
.c1:disabled {
  cursor: not-allowed;
  opacity: 1;
}

.c1:read-only,
.c1:disabled {
  cursor: not-allowed;
  opacity: 1;
}

.c1:disabled {
  background-color: #f1f3f4;
}

.c1::-webkit-input-placeholder {
  opacity: 0;
  -webkit-transition: inherit;
  transition: inherit;
}

.c1::-moz-placeholder {
  opacity: 0;
  -webkit-transition: inherit;
  transition: inherit;
}

.c1:-ms-input-placeholder {
  opacity: 0;
  -webkit-transition: inherit;
  transition: inherit;
}

.c1::placeholder {
  opacity: 0;
  -webkit-transition: inherit;
  transition: inherit;
}

.c1:hover {
  border-color: var(--field-input-hover-border-color);
}

.c1:focus {
  outline: 0;
  border-color: var(--field-input-focus-border-color);
  box-shadow: var(--field-input-box-shadow);
}

.c1:focus::-webkit-input-placeholder {
  opacity: 1;
}

.c1:focus::-moz-placeholder {
  opacity: 1;
}

.c1:focus:-ms-input-placeholder {
  opacity: 1;
}

.c1:focus::placeholder {
  opacity: 1;
}

.c1:hover,
.c1:focus {
  border-width: 2px;
}

.c1:hover ~ label,
.c1:focus ~ label {
  border-width: 2px;
}

.c1:focus,
.c1:not(:placeholder-shown) {
  padding-top: 24px;
  padding-bottom: 8px;
}

.c1:focus ~ label,
.c1:not(:placeholder-shown) ~ label {
  color: var(--field-label-focus-color);
  font-size: 12px;
  padding-top: 8px;
  padding-bottom: 30px;
}

.c1:-webkit-autofill {
  padding-top: 24px;
  padding-bottom: 8px;
}

.c1:-webkit-autofill ~ label {
  color: var(--field-label-focus-color);
  font-size: 12px;
  padding-top: 8px;
  padding-bottom: 30px;
}

.c0 {
  position: relative;
  font-family: 'Open Sans',sans-serif;
  font-size: 16px;
  --field-input-border-color: #bac4ca;
  --field-input-hover-border-color: #8698a2;
  --field-input-focus-border-color: #1ab3ff;
  --field-input-box-shadow: 0 0 0 2px #b3e5ff;
  --field-label-color: #4f6672;
  --field-label-focus-color: #4f6672;
}

<div
  class="c0"
>
  <input
    aria-labelledby="10"
    class="c1"
    disabled=""
    id="9"
    placeholder=" "
    value="Value"
  />
  <label
    class="c2"
    for="9"
    id="10"
  >
    Basic field
  </label>
</div>
`;

exports[`Field should render with error 1`] = `
.c2 {
  position: absolute;
  padding: 16px 20px;
  height: 100%;
  left: 0;
  top: 0;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
  cursor: text;
  pointer-events: none;
  font-weight: 600;
  -webkit-transition: 100ms all ease-in;
  transition: 100ms all ease-in;
  border: 1px solid transparent;
  max-width: 66.66%;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: var(--field-label-color);
}

.c1 {
  height: 54px;
  padding: 0px 20px;
  min-width: 220px;
  width: 100%;
  line-height: 22px;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  background-color: #fff;
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  box-sizing: border-box;
  -webkit-transition: 100ms all ease-in;
  transition: 100ms all ease-in;
  color: #4f6672;
  border-color: var(--field-input-border-color);
}

.c1:-moz-read-only,
.c1:disabled {
  cursor: not-allowed;
  opacity: 1;
}

.c1:read-only,
.c1:disabled {
  cursor: not-allowed;
  opacity: 1;
}

.c1:disabled {
  background-color: #f1f3f4;
}

.c1::-webkit-input-placeholder {
  opacity: 0;
  -webkit-transition: inherit;
  transition: inherit;
}

.c1::-moz-placeholder {
  opacity: 0;
  -webkit-transition: inherit;
  transition: inherit;
}

.c1:-ms-input-placeholder {
  opacity: 0;
  -webkit-transition: inherit;
  transition: inherit;
}

.c1::placeholder {
  opacity: 0;
  -webkit-transition: inherit;
  transition: inherit;
}

.c1:hover {
  border-color: var(--field-input-hover-border-color);
}

.c1:focus {
  outline: 0;
  border-color: var(--field-input-focus-border-color);
  box-shadow: var(--field-input-box-shadow);
}

.c1:focus::-webkit-input-placeholder {
  opacity: 1;
}

.c1:focus::-moz-placeholder {
  opacity: 1;
}

.c1:focus:-ms-input-placeholder {
  opacity: 1;
}

.c1:focus::placeholder {
  opacity: 1;
}

.c1:hover,
.c1:focus {
  border-width: 2px;
}

.c1:hover ~ label,
.c1:focus ~ label {
  border-width: 2px;
}

.c1:focus,
.c1:not(:placeholder-shown) {
  padding-top: 24px;
  padding-bottom: 8px;
}

.c1:focus ~ label,
.c1:not(:placeholder-shown) ~ label {
  color: var(--field-label-focus-color);
  font-size: 12px;
  padding-top: 8px;
  padding-bottom: 30px;
}

.c1:-webkit-autofill {
  padding-top: 24px;
  padding-bottom: 8px;
}

.c1:-webkit-autofill ~ label {
  color: var(--field-label-focus-color);
  font-size: 12px;
  padding-top: 8px;
  padding-bottom: 30px;
}

.c0 {
  position: relative;
  font-family: 'Open Sans',sans-serif;
  font-size: 16px;
  --field-input-border-color: #FF2D1A;
  --field-input-hover-border-color: #FF2D1A;
  --field-input-focus-border-color: #FF2D1A;
  --field-input-box-shadow: 0 0 0 2px #FF8A7F;
  --field-label-color: #4f6672;
  --field-label-focus-color: #E61300;
}

<div
  class="c0"
>
  <input
    aria-labelledby="14"
    class="c1"
    id="13"
    placeholder=" "
    value="Value"
  />
  <label
    class="c2"
    for="13"
    id="14"
  >
    Basic field
  </label>
</div>
`;

exports[`Field should render with success 1`] = `
.c2 {
  position: absolute;
  padding: 16px 20px;
  height: 100%;
  left: 0;
  top: 0;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
  cursor: text;
  pointer-events: none;
  font-weight: 600;
  -webkit-transition: 100ms all ease-in;
  transition: 100ms all ease-in;
  border: 1px solid transparent;
  max-width: 66.66%;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: var(--field-label-color);
}

.c1 {
  height: 54px;
  padding: 0px 20px;
  min-width: 220px;
  width: 100%;
  line-height: 22px;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  background-color: #fff;
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  box-sizing: border-box;
  -webkit-transition: 100ms all ease-in;
  transition: 100ms all ease-in;
  color: #4f6672;
  border-color: var(--field-input-border-color);
}

.c1:-moz-read-only,
.c1:disabled {
  cursor: not-allowed;
  opacity: 1;
}

.c1:read-only,
.c1:disabled {
  cursor: not-allowed;
  opacity: 1;
}

.c1:disabled {
  background-color: #f1f3f4;
}

.c1::-webkit-input-placeholder {
  opacity: 0;
  -webkit-transition: inherit;
  transition: inherit;
}

.c1::-moz-placeholder {
  opacity: 0;
  -webkit-transition: inherit;
  transition: inherit;
}

.c1:-ms-input-placeholder {
  opacity: 0;
  -webkit-transition: inherit;
  transition: inherit;
}

.c1::placeholder {
  opacity: 0;
  -webkit-transition: inherit;
  transition: inherit;
}

.c1:hover {
  border-color: var(--field-input-hover-border-color);
}

.c1:focus {
  outline: 0;
  border-color: var(--field-input-focus-border-color);
  box-shadow: var(--field-input-box-shadow);
}

.c1:focus::-webkit-input-placeholder {
  opacity: 1;
}

.c1:focus::-moz-placeholder {
  opacity: 1;
}

.c1:focus:-ms-input-placeholder {
  opacity: 1;
}

.c1:focus::placeholder {
  opacity: 1;
}

.c1:hover,
.c1:focus {
  border-width: 2px;
}

.c1:hover ~ label,
.c1:focus ~ label {
  border-width: 2px;
}

.c1:focus,
.c1:not(:placeholder-shown) {
  padding-top: 24px;
  padding-bottom: 8px;
}

.c1:focus ~ label,
.c1:not(:placeholder-shown) ~ label {
  color: var(--field-label-focus-color);
  font-size: 12px;
  padding-top: 8px;
  padding-bottom: 30px;
}

.c1:-webkit-autofill {
  padding-top: 24px;
  padding-bottom: 8px;
}

.c1:-webkit-autofill ~ label {
  color: var(--field-label-focus-color);
  font-size: 12px;
  padding-top: 8px;
  padding-bottom: 30px;
}

.c0 {
  position: relative;
  font-family: 'Open Sans',sans-serif;
  font-size: 16px;
  --field-input-border-color: #22C383;
  --field-input-hover-border-color: #22C383;
  --field-input-focus-border-color: #22C383;
  --field-input-box-shadow: 0 0 0 2px #93ECC8;
  --field-label-color: #4f6672;
  --field-label-focus-color: #178257;
}

<div
  class="c0"
>
  <input
    aria-labelledby="12"
    class="c1"
    id="11"
    placeholder=" "
    value="Value"
  />
  <label
    class="c2"
    for="11"
    id="12"
  >
    Basic field
  </label>
</div>
`;
